frontend/src/DialectReportView.tsx

```diff
@@ -14,6 +14,19 @@ export const DialectReportView = ({
 }) => {
   const params = useSearchParams();
 
+  function showAboutBowtie() {
+    const initialUrl = `${window.location.protocol}//${window.location.host}${window.location.pathname}`;
+    const currentUrl = window.location.href;
```

Comment:
Since we are using React, it is better to utilize the available hooks instead of plain javascript. I guess `useLocation` hook is what you need. - https://reactrouter.com/en/main/hooks/use-location

Reply:
I'm not sure if this is the same comment or a related one, but it would be a lot better to not hardcode a list of URLs as well -- surely there's a way of doing this where we only enable it on the component which renders the homepage, and not just do that filtering afterward by URL.

Reply:
> I'm not sure if this is the same comment or a related one, but it would be a lot better to not hardcode a list of URLs as well -- surely there's a way of doing this where we only enable it on the component which renders the homepage, and not just do that filtering afterward by URL.

Okay I will check some other approach and will let you know about the progress.